ert_gui/cli.py

```diff
@@ -1,157 +1,168 @@
+    }
+    return model, simulations_argument
+
+def _setup_ensemble_smoother(args):
+    model = EnsembleSmoother()
+
+    simulations_argument = {
+        "active_realizations": _realizations(args),
+        "target_case": _target_case_name(args, format_mode=False)
+    }
+    return model, simulations_argument
+
+
+def _setup_multiple_data_assimilation(args):
+    model = MultipleDataAssimilation()
+
+    simulations_argument = {
+        "active_realizations": _realizations(args),
+        "target_case": _target_case_name(args, format_mode=True),
+        "analysis_module": _get_analysis_module_name(iterable=False),
+        "weights": args.weights
+    }
+    return model, simulations_argument
+
+def _setup_iterated_ensemble_smoother(args):
+    if args.iterations is not None:
+        ertmodel.setNumberOfIterations(args.iterations)
+
+    model = IteratedEnsembleSmoother()
+    simulations_argument = {
+        "active_realizations": _realizations(args),
+        "target_case": _target_case_name(args, format_mode=True),
+        "analysis_module": _get_analysis_module_name(iterable=True)
+    }
+    return model, simulations_argument
+
+def _get_analysis_module_name(iterable):
+        active_name = ERT.ert.analysisConfig().activeModuleName()
+        modules = ertmodel.getAnalysisModuleNames(iterable)
+
+        if active_name in modules:
+            return active_name
+        elif "STD_ENKF" in modules and not iterable:
+            return "STD_ENKF"
+        elif "RML_ENKF" in modules and iterable:
+            return "RML_ENKF"
+        elif len(modules) > 0:
+            return modules[0]
+
+        return None
+
+def _realizations(args):
+    ensemble_size = ERT.ert.getEnsembleSize()
+    mask = BoolVector(default_value=False, initial_size=ensemble_size)
+    if args.realizations is None:
+        default = _default_realizations(ensemble_size)
+        mask.updateActiveMask(default)
+    else:
+        validator = RangeStringArgument(ensemble_size)
+        validated = validator.validate(args.realizations)
+        if validated.failed():
+            raise ArgumentTypeError("Defined realizations is not within range of ensemble size: {}".format(args.realizations))
+        mask.updateActiveMask(args.realizations)
+    return mask
+
+def _default_realizations(ensemble_size):        
+    return "0-%d" % (ensemble_size - 1)
+
+def _target_case_name(args, format_mode=False):
+    """ @rtype: str """
+    if args.target_case is None:
+        if format_mode:
+            if ERT.ert.analysisConfig().getAnalysisIterConfig().caseFormatSet():
+                return ERT.ert.analysisConfig().getAnalysisIterConfig().getCaseFormat()
+            else:
+                case_name = ertmodel.getCurrentCaseName()
+                return "%s_%%d" % case_name
+        else:
+            case_name = ertmodel.getCurrentCaseName()
+            return "%s_smoother_update" % case_name
+    return args.target_case
```

Comment:
this is painful to read. I would suggest the following (I may have made mistakes, please double check it):
```Python 
if args.target_case is not None:
  return args.target_case
case_name = ertmodel.getCurrentCaseName()
if not format_mode:
  return "{}_smoother_update".format(case_name)
aic = ERT.ert.analysisConfig().getAnalysisIterConfig()
if aic.caseFormatSet():
  return aic.getCaseFormat()
return "{}_%%d".format(case_name)
```

Reply:
absolutely right - been too quick and not refactored the fragments i've copied from gui stuff... will look at your suggestion!